### PR TITLE
SALTO-4927: Allow using parent in serviceUrl

### PR DESCRIPTION
**Pull request by shir-reifenberg**
Allow using parent values with service url through the config.
The format is similar to the one being used with `urlParamsToFields`, for example `api/v1/apps/{_parent.0.id}/schema/{id}`

---

I also change serviceId Filter to use the definitions according to type to avoid creating filter for each config

---
_Release Notes_: 
None

---
_User Notifications_: 
None

### CODE REVIEW

File: packages/adapter-components/src/filters/service_url.ts [hunk 1]
```diff
@@ -13,42 +13,59 @@
...
     .types[instance.elemID.typeName]?.transformation?.serviceUrl
   if (serviceUrl === undefined) {
     return
   }
-  const url = createUrl({ instance, baseUrl: serviceUrl })
+  // parent is ReferenceExpression during fetch, and serialized into full value during deploy
+  const parentValues = instance.annotations[CORE_ANNOTATIONS.PARENT]?.map(
+    (parent: unknown) => (isReferenceExpression(parent) ? parent.value.value : parent)
+  )
+  const parentContext = parentValues?.reduce((
```
Comment (edenhassid) [inline packages/adapter-components/src/filters/service_url.ts:35]: isn't there a way to look for the value only if it is in the url? and not map all the options before?

**Review by edenhassid — APPROVED** [review]
LGTM! 💯 🔥 added two small comments

File: packages/adapter-components/src/fetch/request/utils.ts [hunk 2]
```diff
@@ -18,7 +18,7 @@ import _ from 'lodash'
 import { Values, isPrimitiveValue } from '@salto-io/adapter-api'
 import { ContextParams } from '../../definitions'
 
-export const ARG_PLACEHOLDER_MATCHER = /\{([\w_]+)\}/g
+export const ARG_PLACEHOLDER_MATCHER = /\{([\w_.]+)\}/g
```
[the remark this answers is not in the document]
Comment (shir-reifenberg) [reply packages/adapter-components/src/fetch/request/utils.ts:21]: Talked IRL, will change to \.

[the remark this answers is not in the document]
Comment (shir-reifenberg) [reply packages/adapter-components/src/fetch/request/utils.ts:21]: Actually check again, and there's no need in backslash, "." actually only includes "." 😃

**Pull request merged**